### PR TITLE
feat(azure-blobstorage): Add azure blobstorage connector

### PR DESCRIPTION
**Pull request by ztefanie**
## Description
- Added the Azure Blob Storage connector.
- I did not add any unit tests as there is basically no own logic, but only calling library functions. I compared it with the box connector, which only has a few unit tests for a function generating the path and no e2e tests.
- The azure library was approved by Kalina
- This is a first MVP. We could add e.g. deletion of file (not mentioned in product hub) or other auth methods, if needed. I only added SAS for now, as it seems to be best practice for this kind of requests, but of course I am open to add other auth methods.

Docs PR: https://github.com/camunda/camunda-docs/pull/5974

## Related issues

related to https://github.com/camunda/team-connectors/issues/1064
related to https://github.com/camunda/product-hub/issues/2713

## Checklist

- [x] PR has a **milestone** or the `no milestone` label.

### CODE REVIEW

File: connectors/microsoft/azure-blobstorage/src/main/java/io/camunda/connector/azure/blobstorage/model/core/BlobStorageExecutor.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+import io.camunda.connector.azure.blobstorage.model.response.UploadResponse;
+import io.camunda.document.Document;
+import io.camunda.document.store.DocumentCreationRequest;
+import java.time.Duration;
+import java.util.function.Function;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class BlobStorageExecutor {
+  private final long UPLOAD_TIMEOUT_IN_SECONDS = 30;
```
Comment (sbuettner) [inline connectors/microsoft/azure-blobstorage/src/main/java/io/camunda/connector/azure/blobstorage/model/core/BlobStorageExecutor.java:36]: Can we make this configurable?

File: connectors/microsoft/azure-blobstorage/src/main/java/io/camunda/connector/azure/blobstorage/model/core/BlobStorageExecutor.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    DownloadRetryOptions options = new DownloadRetryOptions().setMaxRetryRequests(3);
+
+    BlobClient blobClient = blobContainerClient.getBlobClient(downloadObject.fileName());
+    BlobDownloadContentResponse contentResponse =
+        blobClient.downloadContentWithResponse(options, null, null, null);
+
+    BinaryData content = contentResponse.getValue();
+
+    log.debug(
+        "Download of file {} to container {} finished with status code: {}",
```
Comment (sbuettner) [inline connectors/microsoft/azure-blobstorage/src/main/java/io/camunda/connector/azure/blobstorage/model/core/BlobStorageExecutor.java:111]: Do we download "to a container"? Or rather "from a container"?

File: connectors/microsoft/azure-blobstorage/src/main/java/io/camunda/connector/azure/blobstorage/model/response/Element.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+import io.camunda.document.Document;
+
+public interface Element {
+  record DocumentContent(Document document) implements Element {}
+
+  record StringContent(String content) implements Element {}
+
+  record JsonContent(Object content) implements Element {}
+}
```
Comment (sbuettner) [inline connectors/microsoft/azure-blobstorage/src/main/java/io/camunda/connector/azure/blobstorage/model/response/Element.java:17]: What about a sealed class?  Then you can also skip the `element` property and have the implementations on the top level. See https://github.com/camunda/connectors/blob/main/connectors/box/src/main/java/io/camunda/connector/box/model/BoxResult.java for an example.

**Review by johnBgood — DISMISSED** [review]
🏅

**Comment by ztefanie** [comment]
> * [ ]  file name property tooltip- "Specify the file name for the downloaded document."

I did rename this to: "Specify the name of the document to be downloaded."

Because this property takes the name of the document that should be downloaded and not what it is named after the download.

**Comment by ztefanie** [comment]
> # QA information
> ## Test Environment
> * Test required in (you can check both):
>   
>   * [x]  SaaS
>   * [x]  SM
> * [ ]  Create a new test for this feature in our [Pre Release Tests](https://modeler.ultrawombat.com/projects/017779a8-9459-4feb-8b47-eeef8b2717ef--connectors-pre-release-test) for @Szik to be able to test
> 
> ## Test Scope
> _Please describe the test scope, happy path, edge cases that come to your mind, and whatever you think might relevant to test_

Happy path: 
- upload a document successfully
- download a document successfully

Errors: invalid auth, invalid container, ...

> 
> ## Test Data
> _Please provide the test data, if needed (files, URLs, code snippets, FEEL expressions)_

This video is quite helpful explaining how to create a storage account, a container and the SAS tokens quite straight forward: https://www.youtube.com/watch?v=ah1XqItWkuc&t=582s

You should have access to Azure Dev Tenant as an Okta application. If not please reach out to IT. Make sure when logging in, that your mail adress ist not your normal `firstname.lastnam@camunda.com` address, but something like this `firstname.lastname@camundaittest.onmicrosoft.com`, because the testing is done on a different tenant.

The docs for this container are already available here: https://docs.camunda.io/docs/next/components/connectors/out-of-the-box-connectors/azure-blob-storage/
So this will hopefully explain how to use it.

**Review by sbuettner — DISMISSED** [review]
Thanks for considering the feedback.

**Review by johnBgood — APPROVED** [review]
